optimum/neuron/models/inference/nxd/backend/cache.py

```diff
@@ -0,0 +1,166 @@
+# coding=utf-8
+# Copyright 2025 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Cache decorator for NeuronX compilation based on `libneuronxla`."""
+
+import hashlib
+import json
+import logging
+import os
+import pathlib
+from contextlib import contextmanager
+from typing import List, Optional, Union
+
+from libneuronxla.neuron_cc_cache import CacheUrl
+from torch_neuronx.xla_impl.trace import HloArtifacts, NeffArtifacts, generate_neff, hlo_compile, setup_compiler_dirs
+
+from .....cache.hub_cache import create_hub_compile_cache_proxy
+from .....utils.patching import patch_everywhere
+
+
+logger = logging.getLogger("Neuron")
+
+
+# Use the same hash function as in transformers_neuronx
+def get_hash_module(hlo_module, flags):
+    # Hashing is pretty fast and neglegible compared to compilation time
+    hash_gen = hashlib.sha256()
+    text = str(hlo_module)
+    if flags is not None:
+        text += flags.replace(" ", "")
+    hash_gen.update(text.encode("utf-8"))
+    hash = str(hash_gen.hexdigest())[:20]
+    return hash
+
+
+@contextmanager
+def neff_cache(cache_dir: Optional[str] = None):
+    """
+    Context manager to patch `torch_neuronx.xla_impl.trace.generate_neff`.
+
+    This temporarily replaces the original function in the `torch_neuronx.xla_impl.trace` module
+    to use a cache for storing and retrieving compiled NEFF files.
+
+    Usage:
+
+        with neff_cache(cache_dir="/path/to/cache"):
+            # Your code that generates NEFF files goes here
+            # The cache will be used to store and retrieve compiled NEFF files
+            # The original function will be restored after exiting the context
+
+    This function uses the `libneuronxla` library to create a compile cache.
+    Each entry in the cache is identified by a hash of the HLO module and its compilation flags.
+
+    Args:
+        cache_dir (`str`, *optional*):
+            Directory to store the cache. If not provided, a default directory will be used.
+    """
+
+    def generate_neff_with_cache(
```

Comment:
That's awesome, did not think of overriding this. Will check how to use it for other traced models :D